src/views/Workflow.vue

```diff
@@ -72,24 +98,49 @@ export default {
      * @type {CylcTree}
      */
     tree: new CylcTree(),
-    isLoading: true
+    isLoading: true,
+    // the widgets added to the view
+    /**
+     * @type {
+     *   Object.<string, string>
+     * }
+     */
+    widgets: {}
   }),
   computed: {
     ...mapState('workflows', ['workflows']),
-    ...mapState('user', ['user'])
+    ...mapState('user', ['user']),
+    treeWidgets () {
+      return Object
+        .entries(this.widgets)
+        .filter(([id, type]) => type === TreeComponent.name)
+        .map(([id, type]) => id)
+    },
+    graphWidgets () {
+      return Object
+        .entries(this.widgets)
+        .filter(([id, type]) => type === GraphComponent.name)
+        .map(([id, type]) => id)
+    },
+    mutationsWidgets () {
+      return Object
+        .entries(this.widgets)
+        .filter(([id, type]) => type === MutationsView.name)
+        .map(([id, type]) => id)
+    }
   },
   created () {
     const vm = this
     EventBus.$on('add:tree', () => {
       const subscriptionId = this.subscribeDeltas()
       // add widget that uses the GraphQl query response
-      this.$refs['workflow-component'].addTreeWidget(`${subscriptionId}`)
+      this.addTreeWidget(subscriptionId)
     })
     EventBus.$on('add:graph', () => {
       // subscribe GraphQL query
       const subscriptionId = this.subscribe('graph')
       // add widget that uses the GraphQl query response
-      this.$refs['workflow-component'].addGraphWidget(`${subscriptionId}`)
+      this.addGraphWidget(`${subscriptionId}`)
```

Comment:
Any reason the `subscriptionId` is passed differently for different components:

```js
this.addTreeWidget(subscriptionId)
this.addGraphWidget(`${subscriptionId}`)
```

Reply:
Ops, I believe it needs to be `${}`, so that it's a `String`, and not a `Number` (from what I recall I had a comparison or some `String` method call somewhere that would cause an issue in runtime). Will fix it later :+1: Good catch!

Reply:
Hmmm, the rest of the code was actually using `number`. So I've actually changed the rest to use a `number`, and added JSdocs as well. Ran the IDE linter and it found no issues in the file related to typing, and tested with `five` locally.